### PR TITLE
Whitelist syscalls linked to CAP_SYS_NICE in default seccomp profile

### LINKED ISSUE #37241: syscalls linked to CAP_SYS_NICE are not whitelisted in default seccomp profile
Bug was reported first in Red Hat bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1588755

With --cap-add=SYS_NICE, seccomp still blocks `set_mempolicy` syscall.
According to the doc (https://docs.docker.com/engine/security/seccomp/) set_mempolicy is gated by CAP_SYS_NICE and should be allowed while using --cap-add=SYS_NICE.

Step to reproduce:
```[builder@dlw11 ~]$ nvidia-docker run  -di --cap-add=SYS_NICE  --name=test --hostname=test   good bash
a10029a063dfea068a256f53f7912671500f1aab1da90f1352c0f51d23a4497d
[builder@dlw11 ~]$ docker exec -it test  bash
[builder@test ~]$ hwloc-bind  --membind node:0 echo hello
hwloc_set_membind 0x00000001 (policy 2 flags 0) failed (errno 1 Operation not permitted)
```
"good" is just a rhel 7.5 docker image

Modifying the default seccomp profile by adding:
```{
                        "names": [
                                "get_mempolicy",
                                "mbind",
                                "name_to_handle_at",
                                "set_mempolicy"
                        ],
                        "action": "SCMP_ACT_ALLOW",
                        "args": [],
                        "comment": "",
			"includes": {
				"caps": [
					"CAP_SYS_NICE"
				]
			},
                        "excludes": {}
                }
```
solves the problem.

[builder@dlw11 ~]$ nvidia-docker run  -di --cap-add=SYS_NICE  --name=test --hostname=test  --security-opt seccomp=/home/builder/seccomp.json  good bash
4ca36edae42f8c48bac6bead4f54d572608e5ea342589132d8f8639b26eeee58
[builder@dlw11 ~]$ docker exec -it test  bash
[builder@test ~]$ hwloc-bind  --membind node:0 echo hello
hello

### PR DESCRIPTION
**Pull request by nvcastet**
* Update seccomp profile to match docker documentation at
  https://docs.docker.com/engine/security/seccomp/
* fixes #37241
Signed-off-by: Nicolas V Castet <nvcastet@us.ibm.com>

### CODE REVIEW

**Comment by thaJeztah** [comment]
ping @justincormack PTAL

**Comment by thaJeztah** [comment]
@nvcastet looks like you need to regenerate some files;

```
16:50:05 The result of go generate ./profiles/seccomp/ differs
16:50:05 
16:50:05  M profiles/seccomp/default.json
16:50:05 
16:50:05 Please re-run go generate ./profiles/seccomp/
16:50:05 
```

**Comment by thaJeztah** [comment]
(and likely squash the two commits)

File: profiles/seccomp/default.json [hunk 1]
```diff
@@ -746,6 +746,23 @@
...
+				"caps": [
+					"CAP_SYS_NICE"
+				]
+			},
+			"excludes": {}
 		}
 	]
-}
\ No newline at end of file
+}
```
Comment (justincormack) [inline profiles/seccomp/default.json]: I think you will need to keep the original file ending without a new line to get the validation to work.

File: profiles/seccomp/seccomp_default.go [hunk 2]
```diff
@@ -630,6 +630,19 @@ func DefaultProfile() *types.Seccomp {
 				Caps: []string{"CAP_SYS_TTY_CONFIG"},
 			},
 		},
+		{
+			Names: []string{
+				"get_mempolicy",
+				"mbind",
+				"name_to_handle_at",
```
Comment (justincormack) [inline profiles/seccomp/seccomp_default.go]: `name_to_handle_at` is nothing to do with `CAP_SYS_NICE` it is gated by `CAP_DAC_READ_SEARCH` and there are other reasons for excluding it.

File: profiles/seccomp/default.json (hunk 1 above)
Comment (nvcastet) [reply profiles/seccomp/default.json]: Will do.

File: profiles/seccomp/seccomp_default.go (hunk 2 above)
Comment (nvcastet) [reply profiles/seccomp/seccomp_default.go]: @justincormack I am totally fine removing it. But in that case the documentation would need to be updated at https://docs.docker.com/engine/security/seccomp/. Search for `name_to_handle_at`, it is mentioned `Already gated by CAP_SYS_NICE.`.

Comment (nvcastet) [reply profiles/seccomp/seccomp_default.go]: Opened docker/docker.github.io#6854

**Comment by nvcastet** [comment]
@thaJeztah Would you know why the `DockerSwarmSuite.TestAPISwarmLeaderElection` test keeps failing?
```
14:20:57 FAIL: docker_api_swarm_test.go:296: DockerSwarmSuite.TestAPISwarmLeaderElection
14:20:57 
14:20:57 [d4789021c34ee] waiting for daemon to start
14:20:57 [d4789021c34ee] daemon started
14:20:57 
14:20:57 [da4194801ef22] waiting for daemon to start
14:20:57 [da4194801ef22] daemon started
14:20:57 
14:20:57 [decc49a95fe76] waiting for daemon to start
14:20:57 [decc49a95fe76] daemon started
14:20:57 
14:20:57 [d4789021c34ee] exiting daemon
14:20:57 assertion failed: error is not nil: Error response from daemon: rpc error: code = DeadlineExceeded desc = context deadline exceeded
14:20:57 [da4194801ef22] daemon started
14:20:57 Attempt #2: daemon is still running with pid 22957
14:20:57 [da4194801ef22] exiting daemon
14:20:57 [decc49a95fe76] exiting daemon
```

**Comment by thaJeztah** [comment]
Looks like that one is marked "flaky"; https://github.com/moby/moby/issues/32673

**Comment by nvcastet** [comment]
@thaJeztah Thanks. Do you know if it is possible to retrigger just the PR jobs that failed (here janky and windowsRS1)?

**Comment by thaJeztah** [comment]
Hm, CI doesn't seem to restart; I asked internally if someone has access to do so

@justincormack PTAL

**Comment by thaJeztah** [comment]
Failure on PowerPC can be ignored;

```
13:56:35 FAIL: docker_cli_daemon_test.go:1800: DockerDaemonSuite.TestDaemonNoSpaceLeftOnDeviceError
13:56:35 
13:56:35 [decb810dce5dc] waiting for daemon to start
13:56:35 [decb810dce5dc] exiting daemon
13:56:35 docker_cli_daemon_test.go:1817:
13:56:35     s.d.Start(c, "--data-root", filepath.Join(testDir, "test-mount"))
13:56:35 /go/src/github.com/docker/docker/internal/test/daemon/daemon.go:203:
13:56:35     t.Fatalf("Error starting daemon with arguments: %v", args)
13:56:35 ... Error: Error starting daemon with arguments: [--data-root /tmp/no-space-left-on-device-test924999940/test-mount]
13:56:35 
```

**Comment by n4ss** [comment]
LGTM !

**Comment by nvcastet** [comment]
@thaJeztah @justincormack Anything else needed to merge this PR?

**Review by justincormack — APPROVED** [review]

**Comment by justincormack** [comment]
Its slightly odd gating these all by `CAP_SYS_NICE` but I think it makes sense.

**Review by thaJeztah — APPROVED** [review]
LGTM

**Comment by thaJeztah** [comment]
@nvcastet will you do a follow up PR in the documentation repo?

**Comment by nvcastet** [comment]
@thaJeztah Documentation PR was created at: docker.github.io#6861

**Comment by thaJeztah** [comment]
Thanks!

Failures look to be flaky tests, so I'll go ahead and merge

**Pull request merged**